scan/src/pages/OracleScriptHomePage.re

```diff
@@ -1,144 +1,344 @@
+           <>
+             <div>
+               <Text
+                 value={request |> Format.iPretty}
+                 weight=Text.Medium
+                 block=true
+                 ellipsis=true
+                 align=Text.Right
+               />
+             </div>
+             <div>
+               <Text
+                 value={"(" ++ (responseTime |> Format.iPretty) ++ "ms)"}
+                 weight=Text.Medium
+                 block=true
+                 color=Colors.gray6
+                 align=Text.Right
+               />
+             </div>
+           </>
+         | _ => <LoadingCensorBar width=70 height=15 />
+         }}
+      </Col.Grid>
+      <Col.Grid col=Col.Two>
+        <div className={CssHelper.flexBox(~justify=`flexEnd, ())}>
+          {switch (oracleScriptSub) {
+           | Data({timestamp}) =>
+             <Timestamp.Grid
+               time=timestamp
+               size=Text.Md
+               weight=Text.Regular
+               textAlign=Text.Right
+             />
+           | _ => <LoadingCensorBar width=100 height=15 />
+           }}
+        </div>
+      </Col.Grid>
+    </Row.Grid>
+  </TBody.Grid>;
+};
+
+let renderBodyMobile =
+    (reserveIndex, oracleScriptSub: ApolloHooks.Subscription.variant(OracleScriptSub.t)) => {
+  switch (oracleScriptSub) {
+  | Data({id, timestamp, description, name, request, responseTime}) =>
+    <MobileCard
+      values=InfoMobileCard.[
+        ("Oracle Script", OracleScript(id, name)),
+        ("Description", Text(description)),
+        ("Requests&\nResponse time", RequestResponse({request, responseTime})),
+        ("Timestamp", Timestamp(timestamp)),
+      ]
+      key={id |> ID.OracleScript.toString}
+      idx={id |> ID.OracleScript.toString}
+    />
+  | _ =>
+    <MobileCard
+      values=InfoMobileCard.[
+        ("Oracle Script", Loading(200)),
+        ("Description", Loading(200)),
+        ("Requests&\nResponse time", Loading(80)),
+        ("Timestamp", Loading(180)),
+      ]
+      key={reserveIndex |> string_of_int}
+      idx={reserveIndex |> string_of_int}
+    />
+  };
 };
 
 [@react.component]
-let make = () =>
-  {
-    let (page, setPage) = React.useState(_ => 1);
-    let pageSize = 10;
+let make = () => {
+  let isMobile = Media.isMobile();
 
-    let oracleScriptsCountSub = OracleScriptSub.count();
-    let oracleScriptsSub = OracleScriptSub.getList(~pageSize, ~page, ());
+  let (page, setPage) = React.useState(_ => 1);
+  let (searchTerm, setSearchTerm) = React.useState(_ => "");
+  let (sortedBy, setSortedBy) = React.useState(_ => LatestUpdate);
 
-    let%Sub oracleScriptsCount = oracleScriptsCountSub;
-    let%Sub oracleScripts = oracleScriptsSub;
+  let pageSize = 10;
+  let oracleScriptsCountSub = OracleScriptSub.count(~searchTerm, ());
+  let oracleScriptsSub = OracleScriptSub.getList(~pageSize, ~page, ~searchTerm, ());
+  let oracleScriptTopPart = OracleScriptSub.getList(~pageSize=6, ~page=1, ~searchTerm="", ());
```

Comment:
Add `//TODO: we'll implement another subscribe function for getting the most requested oracle scripts`